python/hail/dataset.py

```diff
@@ -3419,6 +3419,126 @@ def storage_level(self):
+
+        **Examples**
+
+        Consider the following command which a filter and an annotation to the VDS (we're assuming a split VDS for simplicity):
+        1) an INFO field `AC_HC`, which stores the allele count of high confidence genotypes (DP >= 10, GQ >= 20) for each non-reference allele,
+        2) a filter `HardFilter` that filters all sites with the [GATK suggested hard filters](http://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set):
+            - For SNVs: QD < 2.0 || FS < 60 || MQ < 40 || MQRankSum < -12.5 || ReadPosRankSum < -8.0
+            - For Indels (and other complex): QD < 2.0 || FS < 200.0 || ReadPosRankSum < 20.0
+        >>> vds = hc.read('data/example.vds')
+        >>> vds = vds.annotate_variants_expr(['va.info.AC_HC = gs.filter(g => g.dp >= 10 && g.gq >= 20).callStats(g => v).AC[1:]', 'va.filters = if((v.altAllele.isSNP && (va.info.QD < 2.0 || va.info.FS < 60 || va.info.MQ < 40 || va.info.MQRankSum < -12.5 || va.info.ReadPosRankSum < -8.0)) || (va.info.QD < 2.0 || va.info.FS < 200.0 || va.info.ReadPosRankSum < 20.0)) va.filters.add("HardFilter") else va.filters'])
+
+        If we now export this VDS as VCF, it would produce the following header (for these new fields):
+        ```
+        ##INFO=<ID=AC_HC,Number=.,Type=String,Description=""
+        ```
+
+        This isn't a properly formatted VCF header for the following reasons:
+        1) There is no FILTER entry for `HardFilter`
+        2) Since `AC_HC` has one entry per non-reference allele, its `Number` should be `A`
+        3) `AC_HC` should have a Description
+
+        We can fix this by setting the attributes of these fields:
+        >>> vds = vds.set_va_attribute('va.info.AC_HC','Description','Allele count for high quality genotypes (DP >= 10, GQ >= 20)')
+        >>> vds = vds.set_va_attribute('va.info.AC_HC','Number','A')
+        >>> vds = vds.set_va_attribute('va.filters','HardFilter','This site fails GATK suggested hard filters.')
+
+        Exporting the VDS with the attributes now prints the following header lines:
+        ```
+        ##INFO=<ID=test,Number=A,Type=String,Description="Allele count for high quality genotypes (DP >= 10, GQ >= 20)"
+        ##FILTER=<ID=random,Description="This site is filtered due to poor quality.">
+        ```
+
+        :param str ann_path: Path to variant annotation beginning with `va`.
+        :param str key: Attribute key.
+        :param str value: Attribute value.
+
+        :return: Annotated dataset with the attribute added to the variant annotation.
+        :rtype: :class:`.VariantDataset`
+        """
+
+        return VariantDataset(self.hc, self._jvds.setVAattribute(ann_path,key,value))
+
+    @handle_py4j
+    def get_va_attributes(self, ann_path):
```

Comment:
My instinct is that it should be possible to get attributes from the `Type` object, without needing an extra method. You can certainly do this in the Scala API with `fieldOption`